Don't need to redraw the plugin here, as it's drawn directly by Scintilla.

For Mac, we also have a plugin hack to hide and then re-show the plugin window,
as sometimes the plugin window will go blank and this fixes that issue - see
bug 105056.

diff -r 32e36869f84a layout/generic/nsPluginFrame.cpp
--- a/layout/generic/nsPluginFrame.cpp	Thu Jan 08 23:22:45 2015 -0500
+++ b/layout/generic/nsPluginFrame.cpp	Thu Jan 29 06:20:22 2015 -0800
@@ -1598,6 +1604,23 @@
                            nsRenderingContext& aRenderingContext,
                            const nsRect& aDirtyRect, const nsRect& aPluginRect)
 {
+  // ACTIVESTATE: don't waste time trying to draw again - it's already drawn!
+  if (mInstanceOwner && mInstanceOwner->IsScimozPlugin()) {
+#if defined(XP_MACOSX)
+    // Plugin hack for bug 105056.
+    if (mContent->AttrValueIs(kNameSpaceID_None,
+                              nsGkAtoms::pluginPaintHack,
+                              nsGkAtoms::_true,
+                              eCaseMatters))
+    {
+      mWidget->Show(false);
+      mWidget->Show(true);
+    }
+#endif
+    return;
+  }
+  // ACTIVESTATE: end
+
 #if defined(MOZ_WIDGET_ANDROID)
   if (mInstanceOwner) {
     gfxRect frameGfxRect =
